### ISSUE TITLE
Misc. fitting issues

### ISSUE THREAD

**Issue by jharman25** [BeerTheorySociety/ambigauss#15]
- Gaussian fitting for demo example doesn't work well if you **don't** specify peaks
- Lorentzian fitting struggles if you give the appropriate peak guesses

 Not sure if this is an ambigauss problem or just a general fitting problem, but wanted to note it here.

**Comment by lcwheeler** [comment]
I think that's because the new base class doesn't use the scipy find_peaks_cwt function to identify peaks. Looks like it needs peaks to be specified. I think we'll need to change that in the base class to get it to estimate the peak locations. Maybe run it by @Zsailer?

**Comment by Zsailer** [comment]
> I think that's because the new base class doesn't use the scipy find_peaks_cwt function to identify peaks. Looks like it needs peaks to be specified.

This is true for @jharman25's first point. Currently, you have to specify the number of peaks or it will throw an error. We need to reimplement `find_peaks_cwt`. This should be easy to do.

@jharman25's second point is a little more subtle. The lorentzian model may need more precise guess parameters for the width and amplitudes of the peaks. I think we should include logic somewhere that computes smarter guesses for each parameter. Right now, it sets all parameters equal to 1.

**Comment by lcwheeler** [comment]
Cool, cool. Thanks for the explanation!
